packages/actor-query-operation-path-one-or-more/lib/ActorQueryOperationPathOneOrMore.ts

```diff
@@ -19,17 +20,35 @@ export class ActorQueryOperationPathOneOrMore extends ActorAbstractPath {
   }
 
   public async runOperation(path: Algebra.Path, context: ActionContext): Promise<IActorQueryOperationOutputBindings> {
+    // Such connectivity matching does not introduce duplicates (it does not incorporate any count of the number
+    // of ways the connection can be made) even if the repeated path itself would otherwise result in duplicates.
+    // https://www.w3.org/TR/sparql11-query/#propertypaths
+    if (!context || !context.get('isPathArbitraryLengthDistinct')) {
+      context = context ?
+        context.set('isPathArbitraryLengthDistinct', true) :
+        Map.of('isPathArbitraryLengthDistinct', true);
+      return ActorQueryOperation.getSafeBindings(await this.mediatorQueryOperation.mediate({
+        operation: ActorAbstractPath.FACTORY.createDistinct(path),
+        context,
+      }));
+    }
+
+    context = context.set('isPathArbitraryLengthDistinct', false);
+
     const predicate = <Algebra.OneOrMorePath> path.predicate;
 
     const sVar = path.subject.termType === 'Variable';
     const oVar = path.object.termType === 'Variable';
 
     if (!sVar && oVar) {
       // Get all the results of applying this once, then do zeroOrMore for those
-      const single = ActorAbstractPath.FACTORY.createPath(path.subject, predicate.path, path.object, path.graph);
+      const single = ActorAbstractPath.FACTORY.createDistinct(
```

Comment:
Is this distinct wrapping needed again here?

Reply:
It's not necessary but improves performance, I think. If it is not there and the path without the + has multiple times the same solution as endpoint, there will be started a search with ALP for every time the same solution is encountered.

Reply:
I may be missing something, but due to `isPathArbitraryLengthDistinct` in the context, isn't it already guaranteed that this operation will be wrapped in dinstinct? So AFAICS , this will cause two dinstincts on top of each other.

Reply:
So, what we do is: we apply the path once first, and then we start a search with all the found endpoints with ALP. From all those ALP's together, we have first wrapped this in a distinct, so no solution will be returned twice by the oneOrMore operator without the distinct you are talking about, that's true. But, we also want to start the ALP search with only different values, and that's where this distinct is for.

So this distinct makes sure we don't start an ALP search with twice the same path, the distinct from the AbstractPath (with the context etc) makes sure we dont return the same solution after our whole computation (it filters for example 2 solutions from ALP, obtained with a different startnode).

Reply:
Ah ok, that's clear to me. So for performance-reasons, you do something like `distinct( stuff( distinct(pp) ) )`.

Reply:
That's it!